apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerClientBuilderPartial.RequestIdOnExceptionInterceptor.cs

```diff
@@ -0,0 +1,233 @@
+            AsyncClientStreamingCallContinuation<TRequest, TResponse> continuation) =>
+                WrapException(() =>
+                {
+                    var call = continuation(context);
+                    return new AsyncClientStreamingCall<TRequest, TResponse>(
+                        new SpannerRequestIdStreamWriter<TRequest>(call.RequestStream, context.Options),
+                        WrapExceptionAsync(call.ResponseAsync, context.Options),
+                        WrapExceptionAsync(call.ResponseHeadersAsync, context.Options),
+                        call.GetStatus,
+                        call.GetTrailers,
+                        call.Dispose);
+                },
+                context.Options);
+
+        /// <inheritdoc/>
+        public override AsyncServerStreamingCall<TResponse> AsyncServerStreamingCall<TRequest, TResponse>(
+            TRequest request,
+            ClientInterceptorContext<TRequest, TResponse> context,
+            AsyncServerStreamingCallContinuation<TRequest, TResponse> continuation) =>
+                WrapException(() =>
+                {
+                    var call = continuation(request, context);
+                    var wrappedResponseStream = new SpannerRequestIdStreamReader<TResponse>(call.ResponseStream, context.Options);
+
+                    return new AsyncServerStreamingCall<TResponse>(
+                        wrappedResponseStream,
+                        WrapExceptionAsync(call.ResponseHeadersAsync, context.Options),
+                        call.GetStatus,
+                        call.GetTrailers,
+                        call.Dispose);
+                },
+                context.Options);
+
+        /// <inheritdoc/>
+        public override AsyncDuplexStreamingCall<TRequest, TResponse> AsyncDuplexStreamingCall<TRequest, TResponse>(
+            ClientInterceptorContext<TRequest, TResponse> context,
+            AsyncDuplexStreamingCallContinuation<TRequest, TResponse> continuation) =>
+                WrapException(() =>
+                {
+                    var call = continuation(context);
+                    var wrappedResponseStream = new SpannerRequestIdStreamReader<TResponse>(call.ResponseStream, context.Options);
+                    var wrappedRequestStream = new SpannerRequestIdStreamWriter<TRequest>(call.RequestStream, context.Options);
+
+                    return new AsyncDuplexStreamingCall<TRequest, TResponse>(
+                        wrappedRequestStream,
+                        wrappedResponseStream,
+                        WrapExceptionAsync(call.ResponseHeadersAsync, context.Options),
+                        call.GetStatus,
+                        call.GetTrailers,
+                        call.Dispose);
+                },
+                context.Options);
+
+        /// <summary>
+        /// Gets the request ID from the call options.
+        /// </summary>
+        private static string GetRequestIdFromOptions(CallOptions options)
```

Comment:
Inline this code in `EnrichException` or at least make this method local to `EnrichException`.